### PR TITLE
Tb/responsive publications

### PR DESCRIPTION
**Pull request by meetbhatt23**

### CODE REVIEW

**Comment by meetbhatt23** [comment]
Opened this PR to put my comments from last PR here:
Is this PR meant to be the changes to Publications page like we discussed last week? If yes, then there are a few things missing:
- I don't see the carousel implementation. Use [react slick](https://react-slick.neostack.com/) for that
- The tablet view has a lot of extra space on the right in the publication cards section. Something needs to be done about that. I don't think we should be using a fixed width on these cards at all, and have grid containers define the width for them. Happy to explain this over a call. (image attached)
Screenshot 2020-04-07 at 10 53 10 AM

- Images on the cards are stretched. If we really want the cards to be of a fixed height (and not have heights defined by the image itself) - then you need to add an `object-fit: cover` to your images.
- The alignment of the text with the cards looks quite off compared to the design (images attached)
Screenshot 2020-04-07 at 10 58 52 AM
Screenshot 2020-04-07 at 10 57 39 AM

**Comment by tanvibhakta** [comment]
> The tablet view has a lot of extra space on the right in the publication cards section. Something needs to be done about that. I don't think we should be using a fixed width on these cards at all, and have grid containers define the width for them. Happy to explain this over a call. (image attached)

I'm attaching the comment chain from figma about this here: https://www.figma.com/file/YC5ssdrwYlwUGJDyUNWTXR?node-id=27:1112#22536823

Also refer here, where I was using grids and we had spacing issues (you pointed it out here, https://github.com/obvious/website-v2-source-wordpress/pull/14#issuecomment-603800293) and then we talked about moving to flexbox for just ease of code. I am no longer sure how to structure this bit of code, can you let me know what you would prefer?

**Comment by meetbhatt23** [comment]
Checks are failing, can you check whats wrong?

**Comment by meetbhatt23** [comment]
Also rebase and push?

**Comment by tanvibhakta** [comment]
Yeah I think that's the culprit, sorry

**Comment by meetbhatt23** [comment]
- [ ] Two kinds of arrows are showing up (images attached)
Screenshot 2020-04-09 at 3 51 00 PM
Screenshot 2020-04-09 at 3 50 31 PM

- [ ] On load, the shadow on the right shows up fine. But when I go to the end and come back, its gone (image attached)
Screenshot 2020-04-09 at 3 51 49 PM

- [ ] When the carousel has less items than the width (something that might not happen IRL but we should factor for it) - the arrow still shows which is redundant. Slick should have a way of disabling that, if not then maybe we should. (image attached)
Screenshot 2020-04-09 at 3 54 46 PM


Overall, I feel that the state based logic of showing arrows is too complicated (which is why the shadow initially appears and then doesn't when you come back - I'm guessing), when slick already adds classes to the arrows - we can be smart about it and simply apply an external shadow to an arrow that is active. I think that should cut it.

**Comment by tanvibhakta** [comment]
Also: When the screensize is lesser than the minimum width assigned to the card, the cards appear vertically to one another: 
Screen Shot 2020-04-09 at 4 11 06 PM

**Comment by tanvibhakta** [comment]
Have fixed the first two bugs, thanks for catching them!

> When the carousel has less items than the width (something that might not happen IRL but we should factor for it) - the arrow still shows which is redundant. Slick should have a way of disabling that, if not then maybe we should. (image attached)

This should never happen IRL - we need to rewrite the carousel to have the fifteen most recent articles (It's a todo from Akshay). Either way, react-slick does not give us an easy way to do this at all. Can we put a todo in it and come back to it in a few days, when we start using this carousel in the Publication Article List, etc?

File: package.json [hunk 1]
```diff
@@ -28,13 +28,16 @@
     "gatsby-plugin-remove-serviceworker": "^1.0.0",
     "gatsby-plugin-transition-link": "^1.17.7",
     "gsap": "^3.1.1",
+    "jquery": "^3.4.1",
```
Comment (meetbhatt23) [inline package.json:31]: Why is this here? `react-slick` should take care of that internally yeah?

**Comment by meetbhatt23** [comment]
> Have fixed the first two bugs, thanks for catching them!
> 
> > When the carousel has less items than the width (something that might not happen IRL but we should factor for it) - the arrow still shows which is redundant. Slick should have a way of disabling that, if not then maybe we should. (image attached)
> 
> This should never happen IRL - we need to rewrite the carousel to have the fifteen most recent articles (It's a todo from Akshay). Either way, react-slick does not give us an easy way to do this at all. Can we put a todo in it and come back to it in a few days, when we start using this carousel in the Publication Article List, etc?

This is good. Let's fix the following things
> Also: When the screensize is lesser than the minimum width assigned to the card, the cards appear vertically to one another:
This one

The gradient is too much for mobile, especially when I'm in the middle of the carousel when it shows up on both sides. Design has 32px, we're using 64

File: src/components/ArticleCarousel.js [hunk 2]
```diff
@@ -18,15 +59,63 @@ export function ArticleCarouselCard({ slug, title, datePublished }) {
...
+
+  }
+
   return (
-    <div className="flex py-8 overflow-x-scroll">
+    <div className="relative">
+    <div className="flex flex-row justify-between absolute w-full py-8">
+      {isLeftShown ? <div className="z-10 w-16 h-40 bg-gradient-left" /> : <div className="z-10 w-16 h-40" />}
+      {isRightShown ? <div className="z-10 w-16 h-40 bg-gradient-right" /> : <div className="z-10 w-16 h-40" />}
+    </div>
```
Comment (meetbhatt23) [inline src/components/ArticleCarousel.js:107]: These shouldn't have fixed heights. Make `h-40` here to be `h-full`, same for `ArticleCarouselCard` and add `h-40` to `ArticleCarousel` instead.

**Comment by meetbhatt23** [comment]
Back button has stopped working on desktop because of the following:
Screenshot 2020-04-09 at 5 12 55 PM

You can change state of a component from its child - in that case you will need to use `React.Context`. See instances below
https://github.com/obvious/website-v2-source-wordpress/pull/18/files#diff-bbef1830b0507fd64ace06fd67d6b4e3R23
https://github.com/obvious/website-v2-source-wordpress/pull/18/files#diff-bbef1830b0507fd64ace06fd67d6b4e3R31

File: src/components/ArticleCarousel.js (hunk 2 above)
Comment (meetbhatt23) [reply src/components/ArticleCarousel.js:107]: You might also need to add this for the above to work:
`.slick-track, .slick-list, .slick-slide, .slick-slide > div  {
  height: 100%;
}`

File: package.json (hunk 1 above)
Comment (tanvibhakta) [reply package.json:31]: Good call! Removed this from the dependency list.
